src/libstd/lib.rs

```diff
@@ -43,17 +136,18 @@
 //! [`Iterator`](iter/trait.Iterator.html), which works with the `for`
 //! loop to access collections.
 //!
-//! The common container type, `Vec`, a growable vector backed by an array,
-//! lives in the [`vec`](vec/index.html) module. Contiguous, unsized regions
-//! of memory, `[T]`, commonly called "slices", and their borrowed versions,
-//! `&[T]`, commonly called "borrowed slices", are built-in types for which the
-//! [`slice`](slice/index.html) module defines many methods.
+//! The common container type, `Vec`, a growable vector backed by an
+//! array, lives in the [`vec`](vec/index.html) module. Contiguous,
+//! unsized regions of memory, `[T]`, commonly called "slices", and
+//! their borrowed versions, `&[T]`, commonly called "borrowed
+//! slices", are primitive types [with many implicit
```

Comment:
This whole sentence is a big mouthful. We should consider restructuring it or the whole paragraph. Maybe something like:


--------
The standard library exposes 3 common ways to deal with contiguous regions of memory:

* `Vec<T>` - A heap allocated *vector* that is resizable at runtime.
* `[T; n]` - An inline *array* with a fixed size at compile time.
* `[T]` - A dynamically sized *slice* into any other kind of contiguous storage whether heap-allocated or not.

Slices can only be handled through some kind of *pointer*, and as such come in many flavours such as:
* `&[T]` - *shared slice*
* `&mut [T]` - *mutable slice*
* `Box<[T]>` - *owned slice*